web_validation_dialog/static/src/js/web_validation_dialog.js

```diff
@@ -0,0 +1,112 @@
+odoo.define("web_validation_dialog.ValidationDialog", function(require) {
+    "use strict";
+
+    var core = require('web.core');
+    var FromWidget = require('web.form_widgets');
+    var Dialog = require('web.Dialog');
+    var Model = require('web.DataModel');
+    var framework = require('web.framework');
+    var pyeval = require('web.pyeval');
+
+    var QWeb = core.qweb;
+    var _t = core._t;
+
+    FromWidget.WidgetButton.include({
+        init: function(field_manager, node) {
+            this._super(field_manager, node);
+            this.options = pyeval.py_eval(this.node.attrs.options || '{}');
+            this.is_dialog_security = this.options.security
+            ? this.options.security
+                    : false;
+        },
+        execute_action: function() {
+            var self = this;
+            var exec_action = function() {
+            if (self.node.attrs.confirm && self.is_dialog_security) {
+                Dialog.confirm(self, self.node.attrs.confirm, {
+                    buttons: [
+                              {
+                                  text: _t("Ok"),
+                                  classes: 'btn-primary',
+                                  click: function(){
+                                      this.close();
+                                      self.open_pincode_dialog();
+                                  }
+                              },
+                              {
+                                  text: _t('Cancel'),
+                                  close: true
+                              }]
```

Comment:
Please use proper indentation all across this file.

Reply:
Done